azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -43,6 +43,12 @@ func resourceArmKubernetesCluster() *schema.Resource {
 					return nil
 				}
 
+				networkPolicy := profile["network_policy"].(string)
+
+				if networkPolicy != "calico" {
+					return nil
```

Comment:
Did you mean to return an error here?

I don't think this is required as validation should prevent any other value?

Reply:
Was thinking something like this might work:

"network_policy": {
							Type:     schema.TypeString,
							Optional: true,
							ForceNew: true,
							ValidateFunc: validation.StringInSlice([]string{
								string(containerservice.Calico),
							}, false),
							DiffSuppressFunc: suppress.CaseDifference,
						},

Reply:
Could we not suppress the case difference? We are slowly moving towards making these sorts of fields case sensitive due to the odd case where it matters on the API side.